lntest/itest/lnd_test.go

```diff
@@ -4406,6 +4408,225 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
+			if err != nil {
+				t.Fatalf("unable to get txid: %v", err)
+			}
+			point := wire.OutPoint{
+				Hash:  *txid,
+				Index: chanPoint.OutputIndex,
+			}
+
+			ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+			err = node.WaitForNetworkChannelOpen(ctxt, chanPoint)
+			if err != nil {
+				t.Fatalf("%s(%d): timeout waiting for "+
+					"channel(%s) open: %v", nodeNames[i],
+					node.NodeID, point, err)
+			}
+		}
+	}
+
+	// Create 1 invoice for Bob, which expect a payment from Alice for 1k
+	// satoshis and another of 1k from Carol.
+	const numPayments = 1
+	const paymentAmtSat = 1000
+	_, rHashes, _, err := createPayReqs(
+		net.Bob, 2*paymentAmtSat, numPayments,
+	)
+	if err != nil {
+		t.Fatalf("unable to create pay reqs: %v", err)
+	}
+
+	// Query for routes to pay from Alice to Bob.
+	// We set FinalCltvDelta to 40 since by default QueryRoutes returns
+	// the last hop with a final cltv delta of 9 where as the default in
+	// htlcswitch is 40.
+	payMPP := func(node *lntest.HarnessNode) {
+		routesReq := &lnrpc.QueryRoutesRequest{
+			PubKey:         net.Bob.PubKeyStr,
+			Amt:            paymentAmtSat,
+			FinalCltvDelta: lnd.DefaultBitcoinTimeLockDelta,
+		}
+		ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+		routes, err := node.QueryRoutes(ctxt, routesReq)
+		if err != nil {
+			t.Fatalf("unable to get route from %s: %v",
+				node.Name(), err)
+		}
+
+		addr := [32]byte{1}
+		// Using the node as the source, pay to the invoice from
+		// Bob created above.
+		for _, rHash := range rHashes {
+			r := routes.Routes[0]
+			hop := r.Hops[len(r.Hops)-1]
+			hop.MppTotalAmtMsat = 2 * paymentAmtSat * 1000
+			hop.MppPaymentAddr = addr[:]
+
+			sendReq := &routerrpc.SendToRouteRequest{
+				PaymentHash: rHash,
+				Route:       r,
+			}
+			ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+			resp, err := node.RouterClient.SendToRoute(ctxt, sendReq)
+			if err != nil {
+				t.Fatalf("unable to create payment stream for "+
+					"%s: %v", node.Name(), err)
+			}
+			if resp.Failure != nil {
+				t.Fatalf("received payment error from %s: %v",
+					node.Name(), resp.Failure)
+			}
+		}
+	}
+
+	go payMPP(net.Alice)
+	payMPP(carol)
+
+	printRespJSON := func(resp proto.Message) {
+		jsonMarshaler := &jsonpb.Marshaler{
+			EmitDefaults: true,
+			Indent:       "    ",
+		}
+
+		jsonStr, err := jsonMarshaler.MarshalToString(resp)
+		if err != nil {
+			fmt.Println("unable to decode response: ", err)
+			return
+		}
+
+		fmt.Println(jsonStr)
+	}
+
+	checkPayments := func(node *lntest.HarnessNode) {
+		req := &lnrpc.ListPaymentsRequest{}
+		ctxt, _ = context.WithTimeout(ctxt, defaultTimeout)
+		paymentsResp, err := node.ListPayments(ctxt, req)
+		if err != nil {
+			t.Fatalf("error when obtaining %s payments: %v",
+				node.Name(), err)
+		}
+		if len(paymentsResp.Payments) != numPayments {
+			t.Fatalf("incorrect number of payments, got %v, want %v",
+				len(paymentsResp.Payments), numPayments)
+		}
+
```

Comment:
Should also assert the correctness of the new fields.

Reply:
the new fields aren't exposed in listpayments or listinvoices yet